tests/app/storage/test_encrypted_questionnaire_storage.py

```diff
@@ -29,15 +33,26 @@ def test_encrypted_storage_requires_user_ik(self):
         with self.assertRaises(ValueError):
             EncryptedQuestionnaireStorage("1", None, "pepper")
 
-    def test_store_and_get(self):
-        user_id = "1"
-        user_ik = "2"
-        encrypted = EncryptedQuestionnaireStorage(user_id, user_ik, "pepper")
-        data = "test"
-        encrypted.save(data)
+    def test_store_and_get_without_submitted_at(self):
+        encrypted = EncryptedQuestionnaireStorage(
+            user_id="1", user_ik="2", pepper="pepper"
+        )
+        encrypted.save(data="test")
         # check we can decrypt the data
         self.assertEqual(
-            ("test", QuestionnaireStore.LATEST_VERSION), encrypted.get_user_data()
+            ("test", QuestionnaireStore.LATEST_VERSION, None), encrypted.get_user_data()
+        )
+
+    @freeze_time(datetime.now(tzutc()).replace(second=0, microsecond=0))
+    def test_store_and_get_with_submitted_at(self):
+
+        encrypted = EncryptedQuestionnaireStorage(
+            user_id="1", user_ik="2", pepper="pepper"
+        )
+        encrypted.save(data="test", submitted_at=datetime.now(tz=tzutc()))
+        self.assertEqual(
+            ("test", QuestionnaireStore.LATEST_VERSION, datetime.now(tz=tzutc())),
+            encrypted.get_user_data(),
```

Comment:
`freeze_time` seems unnecessary given that we're setting and getting the time.
```suggestion
    def test_store_and_get_with_submitted_at(self):
		now = datetime.now(timezone.utc)
        encrypted = EncryptedQuestionnaireStorage(
            user_id="1", user_ik="2", pepper="pepper"
        )
        encrypted.save(data="test", submitted_at=now)
        
        self.assertEqual(
            ("test", QuestionnaireStore.LATEST_VERSION, now),
            encrypted.get_user_data(),
```